pallets/cash/src/lib.rs

```diff
@@ -329,6 +353,111 @@ impl<T: Config> pallet_session::SessionManager<SubstrateId> for Module<T> {
     }
 }
 
+/*
+-- Block N --
+changeAuth extrinsic, nextValidators set,  set true
+
+-- Block N + 1 --
+Must end session to get through the already-planned session
+* "ShouldEndSession" should be true bc nextValidators not empty
+* "new_session" returns nextValidators to queue them and deletes the storage
+
+-- Afterwards --
+"ShouldEndSession" when
+* era notices were signed
+
+The next session will finally have the NextValidators, and notice signing can continue
+
+*/
+
+fn intersection_count<T: Ord + Debuggable>(a: Vec<T>, b: Vec<T>) -> usize {
+    let a_count = a.iter().count();
+    let b_count = b.iter().count();
+
+    let mut a_set = BTreeSet::<T>::new();
+    for v in a {
+        a_set.insert(v);
+    }
+
+    let mut b_set = BTreeSet::<T>::new();
+    for v in b {
+        b_set.insert(v);
+    }
+
+    let count = a_set.intersection(&b_set).into_iter().count();
+    println!("a={}, b={}, count={}", a_count, b_count, count);
```

Comment:
I think my recent patch fixes this (the `recover_user_address` versus `recover_address` fix)